# Identifying antibiotic mechanisms of action through differential expression analysis

Final project for 20.440 Spring 2020 with Abhi Aditham.

Microarray data has commonly been used in the past to identify what genes in bacterial genomes antibiotics act on. Using newer RNA-seq datasets we plan to compare insights from both datatypes and see whether we can cluster antibiotics by mechanism of action (MOA) by looking at gene expression only. We hope to identify upstream effector genes and additionally generate a predictor of antibiotic MOA.

## Data
RNA-Seq data from the following paper deposited on the NCBI Gene Expression Omnibus as [GSE110137](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE110137) was used to cluster the gene expression profiles of <em>E. coli</em> treated with different antibiotics.

Zoffmann, S., Vercruysse, M., Benmansour, F. <em>et al.</em> Machine learning-powered antibiotics phenotypic drug discovery. <em>Sci Rep</em> **9,** 5013 (2019).
[![DOIC](https://img.shields.io/badge/DOI-10.1038/s41598--019--39387--9-blue.svg)](https://doi.org/10.1038/s41598-019-39387-9)

## Folder structure
#### `data/`
Contains the count matrix and metadata from Zoffmann <em>et al.</em> used to generate the clustermap.
#### `scripts/`
Contains the main code `heatmap.ipynb` that generates a DataFrame from the RPKM matrix, calculates z-scores, clusters z-scores, and plots.
#### `figs/`
Contains `heatmap1.png` generated by the notebook.
![Clustermap](./figs/heatmap1.png)
Note that some replicates of the same condition do not seem to cluster well.

## Installation
Running `scripts/heatmap.ipynb` generates the figure `heatmap1.png`.

Note seaborn [![PyPI Version](https://img.shields.io/pypi/v/seaborn.svg)](https://pypi.org/project/seaborn/)
is needed over 0.9.0.


 


